### PR TITLE
Form Consumer persists custom field values as donation meta

### LINKED ISSUE #5619: Form Consumer should persist custom field values on the donation
Following the custom form fields documentation, custom field values should be persisted as payment meta on the donation.

## Acceptance Criteria

- [x] Custom field value is persisted as payment meta.

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves #5619

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [ ] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/Form/LegacyConsumer/Commands/SetupFieldPersistance.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		add_action(
+			'give_insert_payment',
+			function( $donationID, $donationData ) use ( $hook ) {
+				$fieldCollection = new FieldCollection( 'root' );
+				do_action( "give_fields_$hook", $fieldCollection, $donationData['give_form_id'] );
+				$fieldCollection->walk(
+					function( $field ) use ( $donationID ) {
+						if ( isset( $_POST[ 'give_' . $field->getName() ] ) ) {
+							$value = wp_strip_all_tags( $_POST[ 'give_' . $field->getName() ], true );
+							give_update_payment_meta( $donationID, '_give_' . $field->getName(), $value );
```
Comment (JasonTheAdams) [inline src/Form/LegacyConsumer/Commands/SetupFieldPersistance.php:24]: I don't think we need to apply the `_give_` prefix to the field meta as it's not actually us that's adding the meta. Other folks are adding these fields, so just using the field name as-is is fine.

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Worked great! I just have one note with regards to the meta key being saved.

**Comment by JasonTheAdams** [comment]
Let's add an Issue as well to give an option for showing the field in the receipt, like happens in the Form Field Manager. I'm thinking:
```php
Fields::text('website')
  ->label('Website')
  ->showInReceipt();
```

File: src/Form/LegacyConsumer/Commands/SetupFieldPersistance.php (hunk 1 above)
Comment (kjohnson) [reply src/Form/LegacyConsumer/Commands/SetupFieldPersistance.php]: See 4c97397

**Review by JasonTheAdams — APPROVED** [review]
Perfect! Let's merge it! 💪

**Pull request merged**